### PR TITLE
migrate: supports INSERT Statement

### PR DESCRIPTION
**Pull request by knwoop**
## WHAT

- support INSERT statement.

## WHY

- to support INSERT statement.
- Currently, Cloud Spanner does not support INSERT in partitioned DML (INSERT was treated as partitioned DML in wrench)
  - ref: https://cloud.google.com/spanner/docs/dml-partitioned#unsupported-features

### CODE REVIEW

File: pkg/spanner/migration.go [hunk 1]
```diff
@@ -171,29 +173,46 @@ func dmlToStatements(filename string, data []byte) ([]string, error) {
 
 func inspectStatementsKind(statements []string) (statementKind, error) {
```
Comment (shuheiktgw) [inline pkg/spanner/migration.go:174]: nit, The current code looks unnecessarily complicated to me. How about refactoring it like below?

```go
func inspectStatementsKind(statements []string) (statementKind, error) {
	var hasDDL, hasDML, hasPartitionedDML bool
	for _, s := range statements {
		switch {
		case isDML(s):
			hasDML = true
		case isPartitionedDML(s):
			hasPartitionedDML = true
		default:
			hasDDL = true
		}
	}

	switch {
	case hasDDL && !hasDML && !hasPartitionedDML:
		return statementKindDDL, nil
	case !hasDDL && hasDML && !hasPartitionedDML:
		return statementKindDML, nil
	case !hasDDL && !hasDML && hasPartitionedDML:
		return statementKindPartitionedDML, nil
	default:
		return "", errors.New("DDL, DML (INSERT), and partitioned DML (UPDATE or DELETE) must not be combined in the same migration file")
	}
}
```

**Review by shuheiktgw — COMMENTED** [review]
Thank you for your PR, @knwoop, sorry that we could not get back to you sooner 🙇 
I left one minor comment, but except that it looks good to me!

**Comment by knwoop** [comment]
> Thank you for your PR, @knwoop, and sorry that we could not get back to you sooner 🙇 I left one minor comment, but except that it looks good to me!

@shuheiktgw Thank you for reviewing my PR! I appreciate your feedback and have addressed the comment.

File: pkg/spanner/migration.go [hunk 2]
```diff
@@ -170,30 +172,38 @@ func dmlToStatements(filename string, data []byte) ([]string, error) {
 }
 
 func inspectStatementsKind(statements []string) (statementKind, error) {
-	kindMap := map[statementKind]uint64{
-		statementKindDDL: 0,
-		statementKindDML: 0,
+	if len(statements) == 0 { // Treat empty files as DDL.
```
Comment (knwoop) [inline pkg/spanner/migration.go:175]: Note: this is treated as a DDL under the existing specifications.

**Review by shuheiktgw — APPROVED** [review]
LGTM! I'll ask for a review from another engineer as well.

**Review by utahta — APPROVED** [review]
LGTM!

**Review by goccy — APPROVED** [review]

**Pull request merged**